api/v1alpha1/hardwareclassification_types.go

```diff
@@ -123,15 +123,68 @@ type Ram struct {
 	MaximumSizeGB int `json:"maximumSizeGB,omitempty"`
 }
 
+// ProfileMatchStatus represents the state of the HardwareClassification
+type ProfileMatchStatus string
+
+const (
+	// ProfileMatchStatusEmpty is the default status value
+	ProfileMatchStatusEmpty ProfileMatchStatus = ""
+
+	// ProfileMatchStatusMatched is the status value when the profile
+	// matches to one of the BareMtalHost.
+	ProfileMatchStatusMatched ProfileMatchStatus = "matched"
+
+	// ProfileMatchStatusUnMatched is the status value when the profile
+	// not matches to one of the BareMtalHost.
+	ProfileMatchStatusUnMatched ProfileMatchStatus = "unmatched"
+)
+
+// ErrorType indicates the class of problem that has caused the HCC resource
+// to enter an error state.
+type ErrorType string
+
+const (
+	// LabelUpdateFailure is an error condition occurring when the
+	// controller is unalble to update label of BareMetalHost.
+	LabelUpdateFailure ErrorType = "label update error"
+
+	// LabelDeleteFailure is an error condition occurring when the
+	// controller is unalble to delete label of BareMetalHost.
+	LabelDeleteFailure ErrorType = "label delete error"
+
+	// FetchBMHListFailure is an error condition occurring when the
+	// controller is unable to fetch BMH from BMO
+	FetchBMHListFailure ErrorType = "fetch BMH from BMO error"
+
+	// ProfileMisConfigured is an error condition occurring when the
+	// extracted profile is empty.
+	ProfileMisConfigured ErrorType = "Empty Profile Error"
+
+	// NoBMHHost is an error condition occurring when the
+	// baremetal host is empty.
+	NoBMHHost ErrorType = "No baremetal host found"
+)
+
 // HardwareClassificationStatus defines the observed state of HardwareClassification
 type HardwareClassificationStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	// ErrorType indicates the type of failure encountered
+	ErrorType ErrorType `json:"errorType,omitempty"`
+
+	// ProfileMatchStatus identifies whether a applied profile is matches or not
+	ProfileMatchStatus ProfileMatchStatus `json:"profileMatchStatus"`
```

Comment:
Did I understand correctly, that this is to indicate whether this HWC matched at least one BMH or not ? What is the use-case behind this ? I'm just trying to understand this.

Reply:
@maelk Yes you are correct, The main motto behind this is once user applies a profile and if an unfortunate error occurred in between then we can update this in the status field to let user know about the error.